classes/check/failingtaskcheck.php

```diff
@@ -87,9 +91,66 @@ public function get_result() : result {
             $status = result::ERROR;
         }
 
+        // Cap the status to the maximum allowed by configuration.
+        $status = $this->get_highest_allowed_warning_level($status);
+
         return new result($status, $this->task->message, '');
     }
 
+    /**
+     * Returns each moodle core result error status as a map of string => integer value, this is used for sorting
+     * alerts when determining the maximum allowed level.
+     *
+     * @return array{na: int, ok: int, info: int, unknown: int, warning: int, error: int, critical: int}
+     */
+    public static function get_integer_values_array() {
+        return array(
```

Comment:
This mapping should already exist elsewhere eg

https://github.com/catalyst/moodle-tool_heartbeat/blob/a187c2c2ae43c9e596b5d43acc7c8345579b439f/classes/checker.php#L35-L51

and

https://github.com/catalyst/moodle-tool_heartbeat/blob/a187c2c2ae43c9e596b5d43acc7c8345579b439f/classes/resultmessage.php#L29-L38

Reply:
I'm happy to use this, but we do lose some information in that some of the result levels get compressed into the same nagios level.

Given most people will just care about the nagios levels, I suspect this is ok.

Reply:
Ah you are right, this should be the moodle api check levels and not the nagios levels. It looks like core itself doesn't have any concept of levels so we do need our own mapping which isn't conflated with nagios.

Sorry for the poor advice :/

I've fixed this myself and another small typo